### PR TITLE
IA-1713 convert CF tasks to TF

### PR DESCRIPTION
**Pull request by jsmigovsky**

### CODE REVIEW

**Review by cdcampbell24 — APPROVED** [review]

File: terraform/application/versions.tf [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 }
-# used for production content bucket 
+# used for production content bucket
 provider "aws" {
-  alias               = "ohio"
-  region              = "us-east-2"
-  version             = "> 2.70"
+  alias  = "ohio"
+  region = "us-east-2"
+  #version             = "> 2.70"
```
Comment (bushong1) [inline terraform/application/versions.tf:11]: Delete

File: terraform/application/tanto.tf [hunk 2]
```diff
@@ -155,3 +155,54 @@ resource "aws_ecs_task_definition" "tanto_remove_cache" {
     }
   }
 }
+
+module "tanto_import_and_sync" {
+  count                        = contains(["staging", "production"], terraform.workspace) ? 0 : 1
```
[the remark this answers is not in the document]
Comment (jsmigovsky) [reply terraform/application/tanto.tf:160]: sure

File: terraform/application/tanto.tf [hunk 3]
```diff
@@ -155,3 +155,54 @@ resource "aws_ecs_task_definition" "tanto_remove_cache" {
...
+  source                       = "USSBA/easy-fargate/aws"
+  version                      = "2.2.1"
+  name                         = "${terraform.workspace}-tanto-import-and-sync-s3"
+  container_image              = "${local.ecr_prefix}downsync:latest"
+  container_command            = ["/bin/bash", "-cxe", "sync.sh $${AWS_ENVIRONMENT}"]
```
[the remark this answers is not in the document]
Comment (jsmigovsky) [reply terraform/application/tanto.tf:165]: I am following the template that was originally designed in CloudFormation.

File: terraform/application/tanto.tf [hunk 4]
```diff
@@ -155,3 +155,54 @@ resource "aws_ecs_task_definition" "tanto_remove_cache" {
...
+      file_system_id = module.efs.efs_id
+      root_directory = "/drupal/boost/"
+      container_path = "/boost/"
+    }
+  ]
+}
+
+data "aws_iam_policy_document" "tanto_import_and_sync" {
+  statement {
+    actions = formatlist("s3:%s", ["ListObject", "GetObject", "ListBucket", "PutObject", "PutObjectAcl", "DeleteObject"])
```
[the remark this answers is not in the document]
Comment (jsmigovsky) [reply terraform/application/tanto.tf:190]: easy enough to change in the future if necessary

File: terraform/application/tanto.tf [hunk 5]
```diff
@@ -155,3 +155,54 @@ resource "aws_ecs_task_definition" "tanto_remove_cache" {
...
+    resources = [
+      "${local.arn_prefix_s3}:::sbagovlower-stash",
```
[the remark this answers is not in the document]
Comment (jsmigovsky) [reply terraform/application/tanto.tf:192]: good catch

File: terraform/application/tanto.tf [hunk 6]
```diff
@@ -155,3 +155,55 @@ resource "aws_ecs_task_definition" "tanto_remove_cache" {
...
   }
 }
+
+# this will NOT be deployed into staging and production, only lower environments
+module "tanto_import_and_sync" {
+  count                        = contains(["staging", "production"], terraform.workspace) ? 0 : 1
+  source                       = "USSBA/easy-fargate/aws"
+  version                      = "~> 2.2"
+  name                         = "${terraform.workspace}-tanto-import-and-sync-s3"
+  container_image              = "${local.ecr_prefix}downsync:latest"
```
Comment (bushong1) [inline terraform/application/tanto.tf:165]: I think you need a slash here.

Comment (jsmigovsky) [reply terraform/application/tanto.tf]: good catch, fixed

**Pull request merged**